.github/workflows/main.yaml

```diff
@@ -1,36 +1,205 @@
 #
 # Generated the prime numbers and generated used by SSH DH Group KEX.
 #
-name: GEN
+# Generating is fast. Max 15 minutes.
+# Validating is slow. 30 hours.
+#
+# Since the GitHub VM have 2 CPU each job will run 2 validating tasks.
+# A job can run for 6 hours... so 12 hours of validation time per job.
+# With 3 jobs for a certain size we should do it.
+#
+# It generates primes for 1024, 1536, 2048, 3072, 4096, 6144, 7680, 8192
+#
+name: CD
 
 # Only run on pull requests and only for Python files.
 on:
-  push:
-    branches: [ master ]
+  # Pull requests are enabled for development.
   pull_request:
-    branches: [ master ]
+    branches: [ main ]
 
+  # https://crontab.guru
+  schedule:
+    # At 01:00 on every 2 weeks on Saturday
+    - cron: '0 1 1-7,15-21 * 6'
 
 jobs:
 
+  prevent_duplicates:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: chevah/auto-cancel-redundant-job@v1
+    - uses: chevah/python-info-action@v1
+
+
   generate:
-    name: 'Size ${{ matrix.size }}'
+    name: 'ssh-keygen -M generate ${{ matrix.size }}'
+    needs: prevent_duplicates
     strategy:
-      fail-fast: false
       matrix:
-        # size: [ 1024, 1536, 2048, 3072, 4096, 6144, 8192]
-        size: [ 1024, 1536, 2048]
+        size: [ 1024, 1536,  2048, 3072, 4096 ]
+        split: [2]
+        include:
+          - size: 6144
+            split: 4
+          - size: 7680
+            split: 10
+          - size: 8192
+            split: 10
+
 
     runs-on: ubuntu-latest
     steps:
 
-    # Make sure we don't have multiple jobs
-    - uses: chevah/auto-cancel-redundant-job@v1
-
     - name: Candidate
       run: |
-        ssh-keygen -G candidate-${{ matrix.size }} -b ${{ matrix.size }}
+
+        ssh-keygen -M generate -O bits=${{ matrix.size }} full.1 &
+        PS1=$!
+        ssh-keygen -M generate -O bits=${{ matrix.size }} full.2 &
+        PS2=$!
```

Comment:
`$!` is quite exotic, so I think it warrants a comment.

Also, why use `PS1` and `PS2` in this way? Like `PROMPT_COMMAND`, these are special env vars.